File: Data Structures & Algorithms/duplicate-integer/submission-1.rs
impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut nums_hash = HashMap::new();
        for num in nums {
            if nums_hash.insert(num, 0).is_some(){
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        if(k==0) return false;
        while(k>0){
            int i=0, j=i+k;
            while(j<nums.size()){
                if(nums[i] == nums[j]){
                    return true;
                }
                i++;j++;
            }
            k--;
        }
        return false;
    }
};